packages/xen-api/index.mjs

```diff
@@ -1409,8 +1436,50 @@ export class Xapi extends EventEmitter {
   }
 }
 
+function cachable(fn, getCache) {
+  return async function (...args) {
+    const cache = getCache(args)
+    if (cache === undefined) {
+      return fn.apply(this, args)
+    }
+
+    const key = this.computeCacheKey(...args)
+    if (cache.has(key)) {
+      return cache.get(key)
+    }
+    const promise = fn.apply(this, args)
+    cache.set(key, promise)
+    try {
+      return promise
+    } catch (error) {
+      cache.delete(key)
+      throw error
+    }
+  }
+}
+
 decorateClass(Xapi, {
-  callAsync: cancelable,
+  call: [
+    cachable,
+    args => {
+      if (typeof args[0] !== 'string') {
```

Comment:
since we use it as a map, I think we should use 
```suggestion
      if ( args[0] instanceof Map) {
```

Reply:
Nope, it's not always a `Map`, it's a object that respect the `Map` interface, which opens the use with other libs like `@isaacs/ttlcache` or `lru-cache`.